### PR TITLE
bpo-42039: Add gc.track and gc.untrack in the gc module

### PR DESCRIPTION
**Pull request by pablogsal**
https://bugs.python.org/issue42039

### CODE REVIEW

File: Doc/library/gc.rst [hunk 1]
```diff
@@ -151,6 +151,15 @@ The :mod:`gc` module provides the following functions:
    be involved in a cycle.  So, for example, if an integer is directly reachable
    from an argument, that integer object may or may not appear in the result list.
 
+.. function:: track(obj)
+
+   Adds the object to the set of container objects tracked by the collector. This function will
+   raise :exc:`ValueError` if the object does not provide gc support.
+
```
Comment (ZackerySpytz) [inline Doc/library/gc.rst:158]: There should be a `versionadded` directive.

File: Doc/library/gc.rst [hunk 2]
```diff
@@ -151,6 +151,15 @@ The :mod:`gc` module provides the following functions:
...
+.. function:: untrack(obj)
+
+   Remove the object from the set of container objects tracked by the collector. This function
+   will raise :exc:`ValueError` if the object does not provide gc support.
```
Comment (ZackerySpytz) [inline Doc/library/gc.rst]: Ditto.

File: Lib/test/test_gc.py [hunk 3]
```diff
@@ -586,6 +586,49 @@ class UserIntSlots(int):
...
+        gc.untrack(x)
+        self.assertFalse(gc.is_tracked(x))
+
+        gc.track(x)
+        gc.track(x)
+        gc.track(x)
+        self.assertTrue(gc.is_tracked(x))
+
+    def test_track_and_untrack_error_cases(self):
+
```
Comment (ZackerySpytz) [inline Lib/test/test_gc.py]:
```suggestion
```

File: Modules/gcmodule.c [hunk 4]
```diff
@@ -1885,6 +1885,54 @@ gc_is_finalized(PyObject *module, PyObject *obj)
     Py_RETURN_FALSE;
 }
 
+/*[clinic input]
+gc.track
+
+    obj: object
+    /
+
+Adds the object to the set of container objects tracked by the collector.
```
Comment (ZackerySpytz) [inline Modules/gcmodule.c]:
```suggestion
Add the object to the set of container objects tracked by the collector.
```

File: Modules/gcmodule.c [hunk 5]
```diff
@@ -1885,6 +1885,54 @@ gc_is_finalized(PyObject *module, PyObject *obj)
...
+[clinic start generated code]*/
+
+static PyObject *
+gc_track(PyObject *module, PyObject *obj)
+/*[clinic end generated code: output=bde3be0acaedaf0d input=1dfbdc582ac876e1]*/
+{
+    if (!PyObject_IS_GC(obj)) {
+        PyErr_Format(PyExc_ValueError, "The object %R has no garbage collector support", obj);
```
Comment (ZackerySpytz) [inline Modules/gcmodule.c]: The PEP 7 style guide states that lines should be limited to 79 characters.

File: Modules/gcmodule.c [hunk 6]
```diff
@@ -1885,6 +1885,54 @@ gc_is_finalized(PyObject *module, PyObject *obj)
...
+
+Remove the object from the set of container objects tracked by the collector.
+[clinic start generated code]*/
+
+static PyObject *
+gc_untrack(PyObject *module, PyObject *obj)
+/*[clinic end generated code: output=c4c94567da0a6043 input=b9f0e2f4df72707a]*/
+{
+    if (!PyObject_IS_GC(obj)) {
+        PyErr_Format(PyExc_ValueError, "The object %R has no garbage collector support", obj);
```
Comment (ZackerySpytz) [inline Modules/gcmodule.c]: Ditto.

**Comment by pablogsal** [comment]
Closing as per https://bugs.python.org/msg378673

**Pull request closed without merging**